python/ray/data/tests/test_dataset_formats.py

```diff
@@ -3102,6 +3103,26 @@ def test_csv_read_filter_no_file(shutdown_only, tmp_path):
         ray.data.read_csv(path)
 
 
+@pytest.mark.skipif(
+    LooseVersion(pa.__version__) < LooseVersion("7.0.0"),
+    reason="invalid_row_handler was added in pyarrow 7.0.0",
+)
```

Comment:
Unfortunately this test does not run on our CI as we use pyarrow 6 there. This test does pass locally though (and does not pass without the fix in `csv_datasource.py`).

Reply:
FYI, Datasets does not support pyarrow >= 7.0.0 due to [this issue](https://github.com/ray-project/ray/issues/22310), and using Datasets with pyarrow 7.0.0+ should fail due to [this check](https://github.com/ray-project/ray/blob/7bec41bd4f7590d45ed2a94625137d1026b5ee94/python/ray/data/_internal/util.py#L36-L62), which is invoked in multiple places.

Reply:
The second check didn't trigger for me with pyarrow 7 and 9